javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,951 @@
+        }
+      }
+      return true
+    }
+    throw new NullPointerException()
+  }
+
+  def replace(oldChar: Char, newChar: Char): _String = {
+    var index = indexOf(oldChar, 0)
+    if (index == -1) {
+      return this
+    }
+    val buffer = Array.ofDim[Char](count)
+    System.arraycopy(value, offset, buffer, 0, count)
+    do {
+      buffer(index += 1) = newChar
+    } while ((index = indexOf(oldChar, index)) != -1);
+    new _String(0, count, buffer)
+  }
+
+  def replace(target: CharSequence, replacement: CharSequence): _String = {
+    if (target == null) {
+      throw new NullPointerException("target should not be null")
+    }
+    if (replacement == null) {
+      throw new NullPointerException("replacement should not be null")
+    }
+    val ts = target.toString
+    var index = indexOf(ts, 0)
+    if (index == -1) return this
+    val rs = replacement.toString
+    if ("" == ts) {
+      val buffer = new StringBuilder(count + (rs.length * (count + 1)))
+      buffer.append(rs)
+      for (i <- 0 until count) {
+        buffer.append(value(offset + i))
+        buffer.append(rs)
+      }
+      return buffer.toString
+    }
+    val buffer = new StringBuilder(count + rs.length)
+    val tl = target.length
+    var tail = 0
+    do {
+      buffer.append(value, offset + tail, index - tail)
+      buffer.append(rs)
+      tail = index + tl
+    } while ((index = indexOf(ts, tail)) != -1);
+    buffer.append(value, offset + tail, count - tail)
+    buffer.toString
+  }
+
+  def startsWith(prefix: _String): scala.Boolean = startsWith(prefix, 0)
+
+  def startsWith(prefix: _String, start: Int): scala.Boolean = {
+    regionMatches(start, prefix, 0, prefix.count)
+  }
+
+  def sub_String(start: Int): _String = {
+    if (start == 0) {
+      return this
+    }
+    if (0 <= start && start <= count) {
+      return new _String(offset + start, count - start, value)
+    }
+    throw new StringIndexOutOfBoundsException(start)
+  }
+
+  def sub_String(start: Int, end: Int): _String = {
+    if (start == 0 && end == count) {
+      return this
+    }
+    if (start < 0) {
+      throw new StringIndexOutOfBoundsException(start)
+    } else if (start > end) {
+      throw new StringIndexOutOfBoundsException(end - start)
+    } else if (end > count) {
+      throw new StringIndexOutOfBoundsException(end)
+    }
+    new _String(offset + start, end - start, value)
+  }
+
+  def toCharArray(): Array[Char] = {
+    val buffer = Array.ofDim[Char](count)
+    System.arraycopy(value, offset, buffer, 0, count)
+    buffer
+  }
+
+  def toLowerCase(): _String = toLowerCase(Locale.getDefault)
+
+  def toLowerCase(locale: Locale): _String = {
+    val result = UCharacter.toLowerCase(locale, this)
+    if (count != result.count) {
+      return result
+    }
+    for (i <- 0 until count if value(offset + i) != result.value(result.offset + i)) {
+      return result
+    }
+    this
+  }
+
+  override def to_String(): _String = this
```

Comment:
This one should be `override def toString(): String = this.asInstanaceOf[_String]`. Given that both `String` and `_String` are the same types at NIR-level, this will be a no-op.